### PR TITLE
add cluster function

### PR DESCRIPTION
**Pull request by double-y**
### Description
add cluster function

resolve https://github.com/exploratory-io/tam/issues/5058

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [x] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/cluster.R [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#' cluster based on vectors
```
Comment (hideaki) [inline R/cluster.R]: Please add a little more explanation on what this function is as a whole.

File: R/cluster.R [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#' @export
+cluster <- function(..., type = "kmeans", n_cluster = 3) {
+  data <- list(...) %>% as.data.frame() %>% as.matrix()
+  na_removed <- data %>% na.omit()
+  if(nrow(na_removed) == 0) {
```
Comment (hideaki) [inline R/cluster.R]: Does k-means work for example when nrow() is 1 or below n_cluster?
If not, we can limit this condition further.

File: R/cluster.R [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    stop("no valid data to cluster")
```
Comment (hideaki) [inline R/cluster.R]: This message is a little too terse to understand what it is quickly. We could be a little more explanatory like this.
"not enough valid data to perform clustering"

File: R/cluster.R [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  }
+  model <- kmeans(na_removed, centers = n_cluster)
+  cluster <- if (ncol(data) == 1) {
```
Comment (hideaki) [inline R/cluster.R]: Please add comment on why we do this distinction on ncol size.

File: R/cluster.R (hunk 1 above)
Comment (double-y) [inline R/cluster.R]: added comments

File: R/cluster.R (hunk 2 above)
Comment (double-y) [inline R/cluster.R]: added a condition

File: R/cluster.R (hunk 3 above)
Comment (double-y) [inline R/cluster.R]: modified the message

File: R/cluster.R (hunk 4 above)
Comment (double-y) [inline R/cluster.R]: added comments

**Pull request merged**